### PR TITLE
Crossword solver in Python

### LINKED ISSUE #594: Crossword Solver in python
The idea behind the Crossword Solver is you type a word with some letters missing, and the program works out all the possible words that would fit with the known letters and spaces.

@kaustubhgupta  Please assign me this task.

### PR DESCRIPTION
**Pull request by aliya-rahmani**
# Description
Added a python AI based on constraint satisfaction algorithm that solves crossword puzzles.
## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Crossword Solver/generate.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        letters = self.letter_grid(assignment)
+
+        # Create a blank canvas
+        img = Image.new(
+            "RGBA",
+            (self.crossword.width * cell_size,
+             self.crossword.height * cell_size),
+            "black"
+        )
+        font = ImageFont.truetype("assets/fonts/OpenSans-Regular.ttf", 80)
```
Comment (kaustubhgupta) [inline Crossword Solver/generate.py:65]: There is no folder named assets/fonts/ .

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Crossword Solver/generate.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        letters = self.letter_grid(assignment)
+
+        # Create a blank canvas
+        img = Image.new(
+            "RGBA",
+            (self.crossword.width * cell_size,
+             self.crossword.height * cell_size),
+            "black"
+        )
+        font = ImageFont.truetype("OpenSans-Regular.ttf", 80)
```
Comment (kaustubhgupta) [inline Crossword Solver/generate.py:65]:
```suggestion
        font = ImageFont.truetype("./Crossword Solver/OpenSans-Regular.ttf", 80)
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by aliya-rahmani** [comment]
@kaustubhgupta I've applied the requested changes, please review it further.

**Review by kaustubhgupta — APPROVED** [review]

**Comment by kaustubhgupta** [comment]
@santushtisharma10 @Kushal997-das

**Comment by santushtisharma10** [comment]
@aliya-rahmani you have copied your code from [here](https://github.com/PLCoster/cs50ai-week3-crossword)

**Comment by kaustubhgupta** [comment]
> @aliya-rahmani you have copied your code from [here](https://github.com/PLCoster/cs50ai-week3-crossword)

yes, fully copied. This PR and issue will be closed @aliya-rahmani

**Pull request closed without merging**